specifications/transactions/onchain/README.md

```diff
@@ -0,0 +1,107 @@
+# On-Chain Data and Transactions
+
+Diem transactions mutate and create state (or resources) within the set of [on-chain modules](https://github.com/diem/diem/tree/master/language/diem-framework/modules), primarily the [Diem Account](https://github.com/diem/diem/blob/master/language/diem-framework/modules/doc/DiemAccount.md). The transaction format is defined in the [Move Adapter Specification](https://github.com/diem/diem/blob/master/specifications/move_adapter/README.md). Most participants of the Diem Payment Network (DPN) will submit SignedTransactions containing a script. Diem stores the scripts in [compiled form](https://github.com/diem/diem/tree/master/language/diem-framework/compiled/legacy/transaction_scripts) and in their [original form](https://github.com/diem/diem/tree/master/language/diem-framework/transaction_scripts). The DPN MainNet only allows this set of pre-registerd scripts to be submitted. Due to the evolving nature of Move and the Move compiler, compiling existing scripts may not result in the form stored in the directory stored above. Hence, it is recommended to use the compiled scripts.
+
+## Peer to Peer Payments and Transaction Metadata
+
+Most transactions will use the [peer_to_peer_with_metadata transaction script](https://github.com/diem/diem/blob/master/language/diem-framework/transaction_scripts/peer_to_peer_with_metadata.move). This single transaction represents all current transfers between two participants and distinguishes the types of transfers via the embedded metadata.
+
+The metadata is represented by the following `Rust` enum encoded in [Binary Canonical Serialization (BCS)](https://github.com/diem/bcs):
+
+```
+enum Metadata {
+  Undefined,
+  GeneralMetadata(GeneralMetadata),
+  TravelRuleMetadata(TravelRuleMetadata),
+  UnstructuredByteMetadata(Option<Vec<u8>>),
+  RefundMetadata(RefundMetadata),
+  CoinTradeMetadata(CoinTradeMetadata),
+}
+```
+
+Note: This is the canonical list and should be referred to in future DIPs so that authors need not reproduce the list in future DIPs.
+
+## Payments Using GeneralMetadata
+
+```
+enum GeneralMetadata {
+   GeneralMetadataV0(GeneralMetadatav0),
+}
+
+struct GeneralMetadataV0 {
+   Option<Vec<u8>> to_subaddress,
+   Option<Vec<u8>> from_subaddress,
+   Option<u64> referenced_event, // Deprecated
+}
+```
```

Comment:
Why is the enum above represented as a rust enum but then this is some mashup of rust and non rust. We should pick a format and stick with how we represent these things.

Reply:
call out the non-rust!